### PR TITLE
Fix "Z3.Base.toBool: illegal `Z3_bool' value" bug for OS X

### LINKED ISSUE #7: Queens4.hs dies with "Z3.Base.toBool: illegal `Z3_bool' value" on OS X
There is a bug on OS X where Z3 uses values other than `1` to represent `True` when evaluating various terms.

Assuming you have nix installed, the following script reproduces the error on OS X.
The script is based on the `Queens4.hs` example.

```haskell
#! /usr/bin/env nix-shell
#! nix-shell -p "haskell.packages.ghc843.ghcWithPackages (pkgs: [pkgs.z3])" -i "cabal exec -- runghc"

module Main where

import Control.Applicative
import Control.Monad ( join )
import Data.Maybe
import qualified Data.Traversable as T

import Z3.Monad

main :: IO ()
main = evalZ3 script >>= \mbSol ->
        case mbSol of
             Nothing  -> error "No solution found."
             Just sol -> putStr "Solution: " >> print sol

script :: Z3 (Maybe [Integer])
script = do
  q1 <- mkFreshIntVar "q1"
  q2 <- mkFreshIntVar "q2"
  q3 <- mkFreshIntVar "q3"
  q4 <- mkFreshIntVar "q4"
  _1 <- mkInteger 1
  _4 <- mkInteger 4
  -- the ith-queen is in the ith-row.
  -- qi is the column of the ith-queen
  assert =<< mkAnd =<< T.sequence
    [ mkLe _1 q1, mkLe q1 _4  -- 1 <= q1 <= 4
    , mkLe _1 q2, mkLe q2 _4
    , mkLe _1 q3, mkLe q3 _4
    , mkLe _1 q4, mkLe q4 _4
    ]
  -- different columns
  assert =<< mkDistinct [q1,q2,q3,q4]
  -- avoid diagonal attacks
  assert =<< mkNot =<< mkOr =<< T.sequence
    [ diagonal 1 q1 q2  -- diagonal line of attack between q1 and q2
    , diagonal 2 q1 q3
    , diagonal 3 q1 q4
    , diagonal 1 q2 q3
    , diagonal 2 q2 q4
    , diagonal 1 q3 q4
    ]
  -- check and get solution
  fmap snd $ withModel $ \m ->
    catMaybes <$> mapM (evalInt m) [q1,q2,q3,q4]
  where mkAbs x = do
          _0 <- mkInteger 0
          join $ mkIte <$> mkLe _0 x <*> pure x <*> mkUnaryMinus x
        diagonal d c c' =
          join $ mkEq <$> (mkAbs =<< mkSub [c',c]) <*> (mkInteger d)
```

### PR DESCRIPTION
**Pull request by xcthulhu**
Closes #7

### CODE REVIEW

**Comment by xcthulhu** [comment]
Ping @IagoAbal ?

**Comment by IagoAbal** [comment]
> Ping @IagoAbal ?

Sorry, I wrote a review many days ago but I forgot to submit it!

File: src/Z3/Base.hs [hunk 1]
```diff
@@ -3192,9 +3192,8 @@ toResult lb
 -- 'Foreign.toBool' should be OK but this is more convenient.
 toBool :: Z3_bool -> Bool
 toBool b
-    | b == z3_true  = True
     | b == z3_false = False
-    | otherwise     = error "Z3.Base.toBool: illegal `Z3_bool' value"
+    | otherwise = True
```
[the remark this answers is not in the document]
Comment (xcthulhu) [reply src/Z3/Base.hs]: Sure! This issue only comes up in OS X (and not linux for some reason I haven't researched.  I'll add a note that this is a work around.

[the remark this answers is not in the document]
Comment (IagoAbal) [reply src/Z3/Base.hs]: Thanks, I will merge it as soon as you have added that note.

**Comment by IagoAbal** [comment]
Hi @xcthulhu, due to issue #9 I'll make a release tomorrow. If you want to get this patch into that release, please update your pull request as we discussed. Thank you!

**Comment by xcthulhu** [comment]
Hey, not sure if I am in time for the release but I have added the comment as requested!

**Comment by IagoAbal** [comment]
I amended the commit and --force pushed... so the commit is actually 17869ff18003e58544c15de458877468d2627fe6.

**Pull request merged**